### PR TITLE
Open image file on Windows

### PR DESCRIPTION
**Pull request by icm7216**
In Windows environment IO class has two modes, that is in text mode and binary mode.
Image files must be loaded in binary mode.

### Before applying the patch. 

Image files are not displayed in the IRuby on Windows.
iruby_before

### After applying the patch. 

Image file is displayed.
iruby_after

### CODE REVIEW

File: lib/iruby/display.rb [hunk 1]
```diff
@@ -287,7 +287,11 @@ def format(mime = nil, &block)
       match {|obj| obj.respond_to?(:path) && File.readable?(obj.path) }
       format do |obj|
         mime = MimeMagic.by_path(obj.path).to_s
-        [mime, File.read(obj.path)] if SUPPORTED_MIMES.include?(mime)
+        if RUBY_PLATFORM =~ /mswin(?!ce)|mingw|cygwin/
+          [mime, File.binread(obj.path)] if SUPPORTED_MIMES.include?(mime)
+        else
+          [mime, File.read(obj.path)] if SUPPORTED_MIMES.include?(mime)
+        end
```
Comment (zverok) [inline lib/iruby/display.rb]: Well, probably `#binread` is appropriate for binary mime types on all platforms?..

Comment (icm7216) [inline lib/iruby/display.rb]: Only Windows platform uses `#binread`.
```
if RUBY_PLATFORM =~ /mswin(?!ce)|mingw|cygwin/
```

**Comment by icm7216** [comment]
I added mime types check, that apply only 'image/png' and 'image/jpeg' on windows platform.

**Comment by kojix2** [comment]
Thank you @icm7216 
I agree with this change.
What do you think? @mrkn

**Comment by mrkn** [comment]
OMG!  I didn't know the feature for displaying an IO object!

I think this feature is evil.  We must drop it.
And we should provide the alternative feature to displaying the content of a file as an image.

**Comment by kojix2** [comment]
Yes. It should be so.
But, you know It is hard to create and maintain image classes for Ruby.
https://github.com/mrkn/IMF

**Comment by mrkn** [comment]
@kojix2 we don't need such a thing just for displaying a file as an image.
It can be done by providing a new method that takes a path of the file, reads the contents, and calls `IRuby.display` with an appropriate mime type.
If such a method is provided as `IRuby.display_file`, we can do:

```ruby
IRuby.display_file(io.path)
```

instead of just displaying an IO.

**Comment by kojix2** [comment]
@mrkn I get the point.

What about the following example?
```ruby
IRuby.display(file: "foo.png")
```

like ruby-gnome2
```ruby
require 'gtk3'
GdkPixbuf::Pixbuf.new(file: "logo-32x32.png")
```

It does not mean that I recommend writing that way.
I just want to hear your opinion about other possibilities.

**Comment by mrkn** [comment]
@kojix2 I want to make a method simple as possible.

**Comment by kojix2** [comment]
@icm7216 
Thank you very much for the great pull request.
This allowed us to find problems with IRuby

**Pull request closed without merging**